c7n/resources/ebs.py

```diff
@@ -454,6 +472,104 @@ def process(self, resources, event=None):
         return self.get_matching_aliases(resources)
 
 
+@Snapshot.filter_registry.register('invalid-kms')
+class KmsKeyActiveSnapshot(Filter):
+    """
+    Filter to ignore snapshots that have an active KMS Key.
+
+    This filter is 'true' by default.
+
+    :example:
+
+    implicit with no parameters, 'true' by default
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-snapshots-with-missing-keys
+                resource: ebs-snapshot
+                filters:
+                  - invalid-kms
+
+    :example:
+
+    explicit with parameter
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-snapshots
+                resource: ebs-snapshot
+                filters:
+                  - type: invalid-kms
+                    value: false
+
+    """
+
+    schema = type_schema('invalid-kms', value={'type': 'boolean'})
+
+    def get_permissions(self):
+        return self.manager.get_resource_manager('ec2').get_permissions()
+
+    def validate(self):
+        if not isinstance(self.data.get('value', True), bool):
```

Comment:
This is redundant validation.  jsonschema will return an error at the validation step if `value` is not boolean as defined by your schema.